sources/packages/api/src/database/entities/application.model.ts

```diff
@@ -185,4 +186,22 @@ export class Application extends RecordDataModel {
     name: "pir_denied_other_desc",
   })
   pirDeniedOtherDesc: string;
+
+  @RelationId((application: Application) => application.coeDeniedReason)
+  coeDeniedId?: number;
+
+  @ManyToOne(() => COEDeniedReason, {
```

Comment:
I haven't seen any advantage in using the relationship Object here. i.e while persisting/saving application we just assign an id to it. And in case of other again we just save the id and save coeDeniedOtherDesc as well. 
So can this reference just be 
@Column({
    name: "coe_denied_id",
    nullable: true,
  })
  coeDeniedReason: number;

we can skip the join in select as well if we just reference the column.

Correct me if I am wrong and this relation object is required for a purpose.

Reply:
The advantage is to load an object and the related object with one command.
For instance, the below code will load the application and the related objects. It is also helpful while creating joins and loading some data from the main object and some data from related objects. Does it make sense?
```
await applicationRepository.findOne(testApplication.id, { relations: ["pirProgram", "offering", "location"] });
```

Reply:
Yes, but if there is no requirement or need to load the related object, ORM relationship is not of any use. ORM Object mapping is required only when you read/edit the object data of relation object. In our case form is doing that part with the options list api and passes on the FK to persist.

Reply:
another advantage, I felt was while saving, we can save the object directly like below
![image](https://user-images.githubusercontent.com/77353155/132597452-ebcd7af5-a812-4c1e-ae20-c4ca9f1a6348.png)

Reply:
@ann-aot but here we are not leveraging the benefit of using relational Object.

Reply:
This is not a blocker, but we can definitely talk about it outside the scope of the PR about using object only when it is required.